### PR TITLE
Sealed box support

### PR DESCRIPTION
**Pull request by whs**
This pull request added support for [sealed box](https://download.libsodium.org/doc/public-key_cryptography/sealed_boxes.html).

Related: #81.

### CODE REVIEW

**Comment by thatch45** [comment]
Sorry I missed you, I have been on vacation! Assuming this looks good I will get a release cut with this added.

File: doc/topics/raw_sealed.rst [hunk 1]
```diff
@@ -0,0 +1,46 @@
+=========================
+Raw Sealed Box Encryption
+=========================
+
+.. note::
+
+    While these routines are perfectly safe, higher level convenience
+    wrappers are under development to make these routines easier.
```
Comment (thatch45) [inline doc/topics/raw_sealed.rst]: Loos like you got he wrappers in this PR, so should this note change?

File: libnacl/sealed.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import libnacl
+import libnacl.public
+import libnacl.dual
+
+class SealedBox(object):
```
Comment (thatch45) [inline libnacl/sealed.py]: This should inherit from libnacl.base.BaseKey so it gets all the helper funcs for key retrieval and saving etc

File: tests/unit/test_raw_public.py [hunk 3]
```diff
@@ -43,7 +43,7 @@ def test_box(self):
         pk3, sk3 = libnacl.crypto_box_keypair()
         pk4, sk4 = libnacl.crypto_box_keypair()
         enc_msg2 = libnacl.crypto_box(msg, nonce2, pk4, sk3)
-        self.assertNotEqual(msg, enc_msg)
+        self.assertNotEqual(msg, enc_msg2)
```
Comment (thatch45) [inline tests/unit/test_raw_public.py]: nice catch! Thanks!

**Comment by thatch45** [comment]
This looks good @whs, but I have made a few comments for some changes

File: doc/topics/raw_sealed.rst (hunk 1 above)
Comment (whs) [inline doc/topics/raw_sealed.rst]: I copied this from other raw_* documentations. Maybe it is unnecessary for the sealed commands.

File: libnacl/sealed.py (hunk 2 above)
Comment (whs) [inline libnacl/sealed.py]: Sealed box doesn't have special keys, it support the same key objects as normal box (as it use box underneath). I see that libnacl.public.Box also doesn't have those key management functions inherited so I don't think SealedBox would be any different.

**Comment by thatch45** [comment]
Looks good! Thanks!

**Pull request merged**